src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -47,28 +47,43 @@ FailureDetector::FailureDetector(ModuleParams *parent) :
 {
 }
 
-bool FailureDetector::resetStatus()
+bool FailureDetector::resetAttitudeStatus()
 {
-	bool status_changed = _status != FAILURE_NONE;
-	_status = FAILURE_NONE;
+
+	int attitude_fields_bitmask = _status & (FAILURE_ARM_ESCS - 1);
+	bool status_changed(false);
+
+	if (attitude_fields_bitmask > FAILURE_NONE) {
+		_status &= ~attitude_fields_bitmask;
+		status_changed = true;
+	}
 
 	return status_changed;
 }
 
 bool
 FailureDetector::update(const vehicle_status_s &vehicle_status)
 {
+
 	bool updated(false);
 
 	if (isAttitudeStabilized(vehicle_status)) {
-		updated = updateAttitudeStatus();
+		updated |= updateAttitudeStatus();
 
 		if (_param_fd_ext_ats_en.get()) {
 			updated |= updateExternalAtsStatus();
 		}
 
 	} else {
-		updated = resetStatus();
+		updated |= resetAttitudeStatus();
+	}
+
+	if (areEscSmart()) {
```

Comment:
This does not do what I would expect from the method name.
Rather use `_sub_esc_status.updated()` directly.

Reply:
Agreed